src/flows/pipes/QueryBuilder/context.tsx

```diff
@@ -268,20 +266,35 @@ export const QueryBuilderProvider: FC = ({children}) => {
       ? EXTENDED_TAG_LIMIT
       : DEFAULT_TAG_LIMIT
 
-    // TODO: Use the `v1.tagKeys` function from the Flux standard library once
-    // this issue is resolved: https://github.com/influxdata/flux/issues/1071
-    query(
-      `${_source}
-              |> range(${formatTimeRangeArguments(range)})
-              |> filter(fn: (r) => ${tagString})
-              |> keys()
-              |> keep(columns: ["_value"])
-              |> distinct()${searchString}${previousTagString}
-              |> filter(fn: (r) => r._value != "_time" and r._value != "_start" and r._value !=  "_stop" and r._value != "_value")
-              |> sort()
-              |> limit(n: ${limit})`,
-      scope
-    )
+    let queryText = `${_source}
+    |> range(${formatTimeRangeArguments(range)})
+    |> filter(fn: (r) => ${tagString})
+    |> keys()
+    |> keep(columns: ["_value"])
+    |> distinct()${searchString}${previousTagString}
+    |> filter(fn: (r) => r._value != "_time" and r._value != "_start" and r._value !=  "_stop" and r._value != "_value")
+    |> sort()
+    |> limit(n: ${limit})`
+
+    if (
+      data.buckets[0].type !== 'sample' &&
+      isFlagEnabled('MeasurementFieldsNoTime')
```

Comment:
I'm curious, does this need to be defined somewhere, or will it just get a default value of `false`?

Could this feature flag be named something more informative to indicate that it affects the query builder? Like `QueryBuilderUseMetadataCaching` or something like that?